### ISSUE TITLE
Crash on share with non existing preset foldet

### ISSUE THREAD

**Issue by PulsarFX** [nextcloud/android#11722]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

Prepare:
1. Share image to nextcloud and choose a nested folder
2. Move that folder on nextcloud two levels up or rename the parent

Act:
1. Share image to nextcloud
2. Preselected is the previous folder (which does not exist anymore)
3. Try to go a level up with arrow at top left
4. Crash

### Expected behaviour

Fall back to an exisiting folder

### Cause of error
```java
Exception in thread "Thread-8" java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.owncloud.android.datamodel.OCFile.getRemotePath()' on a null object reference
    at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:230)
    at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:399)
    at java.lang.Thread.run(Thread.java:920)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30240290`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Model: `SM-G970F`

### Firmware
* SDK: `31`
* Release: `12`
* Incremental: `G970FXXSGHWC1`

### Android version

12

### Device brand and model

Samsung s10e

### Stock or custom OS?

Stock

### Nextcloud android app version

30240290

### Nextcloud server version

25.0.6

### Using a reverse proxy?

I don't know

**Comment by joshtrichards** [comment]
Duplicate of #11207

**Comment by joshtrichards** [comment]
Fixed in #11410 pending the next release.
